PetGame/Services/PetService.cs

```diff
@@ -200,5 +211,196 @@ public bool DeletePet(ulong petId, ulong ownerId)
+            if (PastActivities.Count == 0)
+            {
+                HoursSinceLastActivity = HoursToCheck;
+            }
+            else
+            {
+                //get the last activity
+                Activity LastActivity = PastActivities[PastActivities.Count - 1];
+
+                //check the number of hours since the last activity
+                HoursSinceLastActivity = DateTime.Now.Hour - LastActivity.Timestamp.Hour;
+
+                //check the number of minute since the last activity
+                int MinutesSinceLastActivity = 0;
+                MinutesSinceLastActivity = DateTime.Now.Minute - LastActivity.Timestamp.Minute;
+
+                //if it's been more than 5 minutes, the user can perform another action
+                //otherwise, leave the default 5 min value in place
+                if (MinutesSinceLastActivity > 5)
+                {
+                    MinutesTillNextAction = 0;
+                }
+            }
+
+            //multiply HappinessDecrease by the number of hours to get the
+            //happiness percentage
+            happinessPercentage = (HoursSinceLastActivity * (-1 * HappinessDecreasePerHour));
+
+            //ensure that happiness cannot be <0 or >1
+            if (happinessPercentage < 0.0)
+            {
+                happinessPercentage = 0.0;
+            }
+            else if (happinessPercentage > 1.0)
+            {
+                happinessPercentage = 1.0;
+            }
+
+            //ensure that hunger cannot be <0 or >1
+            if (hungerPercentage < 0.0)
+            {
+                hungerPercentage = 0.0;
+            }
+            else if (hungerPercentage > 1.0)
+            {
+                hungerPercentage = 1.0;
+            }
+
+            //compile the data into a new PetStatus object
+            //return the new object
+            return new PetStatus() { Pet = toReturn, Hunger = hungerPercentage, Happiness = happinessPercentage,
+                TimeOfNextAction = DateTime.Now.AddMinutes(MinutesTillNextAction)};
+        }//end of function
+
+        public IEnumerable<PetStatus> GetPetStatusList(ulong UserId)
+        {
+            List<ulong> PetIdList = new List<ulong>();
+            List<PetStatus> PetStatusList = new List<PetStatus>();
+
+            using (var conn = sqlManager.EstablishDataConnection)
+            {
+                var cmd = conn.CreateCommand();
+
+                cmd.CommandText = @"SELECT Pet.PetId FROM PET WHERE Pet.UserId = @UserId;";
+
+                cmd.Parameters.AddWithValue("@UserId", $"{UserId}");
+
+                using (var reader = cmd.ExecuteReader())
+                {
+                    while (reader.Read())
+                    {
+                        PetIdList.Add((ulong) reader.GetInt64(0));
+                    }
+                }
+
+                if (PetIdList.Count == 0)
+                {
+                    return null;
```

Comment:
Why would this return null, and not an empty list? I don't think you should do that.